packages/slate-react/src/hooks/use-children.tsx

```diff
@@ -50,7 +50,11 @@ const useChildren = (props: {
     const range = Editor.range(editor, p)
     const sel = selection && Range.intersection(range, selection)
 
-    const ds = decorations.filter(dec => Range.intersection(dec, range))
+    const ds = decorations.reduce<Range[]>((acc, dec) => {
+      const intersection = Range.intersection(dec, range)
+      if (intersection) acc.push(intersection)
+      return acc
+    }, [])
```

Comment:
Good call on the `useDecorate()`, I missed that it was left over. 

I didn't switch to the flapMap yet, because I'm not sure about the performance. It looks much nicer and I would expect it to be optimized better over time in browsers, but my current result on comparison benchmarks shows it running much slower for these cases. What do you think?

https://www.measurethat.net/Benchmarks/Show/14729/0/flatmap-vs-filtermap-vs-reduce-v4#latest_results_block

Reply:
Yeah, the benchmark would be a tiebreaker here then, lets leave it. Thanks for that.